csharp/ql/src/semmle/code/csharp/Type.qll

```diff
@@ -111,7 +111,13 @@ class ValueOrRefType extends DotNet::ValueOrRefType, Type, Attributable, @value_
   }
 
   /** Gets the immediate base class of this class, if any. */
-  Class getBaseClass() { extend(this, getTypeRef(result)) }
+  Class getBaseClass() {
+    extend(this, getTypeRef(result))
+    or
+    not extend(this, _) and
+    not this instanceof ObjectType and
+    result instanceof ObjectType
```

Comment:
Why is this `result instanceof ObjectType` and not `result = ObjectType`?

Reply:
The latter is not valid QL syntax, as the right-hand-side is a QL class and not an expression. `result = any(ObjectType ot)` would work, but `instanceof` is better IMO. Note that the extractor will guarantee that there is indeed only one element inside `ObjectType`.